app/utils/callback/lava.py

```diff
@@ -676,6 +672,49 @@ def _find_new_end_line(end_line_number, log, test_case_id):
     return end_line_number
 
 
+def _unpack_results(results):
+    return {test_suite: yaml.load(results_yaml, Loader=yaml.CLoader)
+            for test_suite, results_yaml in results.items()}
+
+
+def _filter_log_data(log, filters_funcs):
+    for filter_func in filters_funcs:
+        log[:] = filter(filter_func, log)
```

Comment:
Well that's all fun design but it can really all be done with a couple of lines and the `lava_filters.py` module could be dropped:
```suggestion
def _filter_log_data(log, filters_funcs):
    def _filter_level(log_line):
        return log_line['lvl'] == 'target'
    def _filter_signals(log_line):
        return not LAVA_SIGNAL_PATTERN.match(log_line['msg'])
    for filter_func in [_filter_level, _filter_signals]:
        log[:] = filter(filter_func, log)
```
Do we want to reuse these filter functions anywhere else via the `lava_filters` module?

Reply:
There were two ideas behind that

> Seems OK but maybe make the LAVA message regex more inclusive, potentially anything starting with `LAVA_` or a set of `LAVA_*` patterns. Also I'm not sure why the LAVA filters got their own module, refactoring the design should rather be done as part of moving kernelci-backend functionality to client-side.

I don't quite get how LAVA callback processing could be moved to client side.

Reply:
> Well that's all fun design but it can really all be done with a couple of lines and the `lava_filters.py` module could be dropped:
> 
> Do we want to reuse these filter functions anywhere else via the `lava_filters` module?

There were couple of ideas when moving functions to the separate module:

- I wanted to separate the actual filters from the code that parses logs with them. 
- Prepare one convenient place to add new filters without browsing through `lava.py` code with clean rules and automatic filters execution.

Module creation had nothing to do with code reuse, rather with code separation.

Reply:
This could be moved client-side by having a separate listening service to handle LAVA callbacks, and then push results to the common backend API.

About the separate module, I'm not sure the overhead of having more code to maintain it worth it especially as I don't know if we'll be adding many more filters of this kind.  But that's OK, it's more a matter of style.